### PR TITLE
WAF-294 Category filters

### PR DESCRIPTION
**Pull request by rhudea**
Support category filter
Made `npm run dev` run again

### CODE REVIEW

File: src/context/products.tsx [hunk 1]
```diff
@@ -78,11 +78,11 @@ const ProductsContext = createContext<{
   setListViewType: (viewType: string) => void;
   resolveCartId?: () => Promise<string | undefined>;
   refreshCart?: () => void;
-  addToCart?: (
+  addToCart: (
```
Comment (yugandhar02) [inline src/context/products.tsx:81]: The addToCart api needs to be fixed in separate PR
It doesn't return error in case of network issues

**Review by yugandhar02 — APPROVED** [review]

File: src/context/products.tsx (hunk 1 above)
Comment (rhudea) [reply src/context/products.tsx:81]: All the addToCart changes are related to making `npm run dev` to be able to develop the functionality.
